app/controllers/account/pages_controller.rb

```diff
@@ -10,10 +10,11 @@ def show
   private
 
   def resource
-    Page.find(params[:id])
+    company = current_user.companies.find(params[:company_id])
+    company.pages.find(params[:id])
```

Comment:
why not just simplify this to `collection.find(params[:id])` ?

Reply:
+1